anthos-bm-gcp-terraform/docs/one_click_install.md

```diff
@@ -0,0 +1,133 @@
+
+### Step by step guide
+
+1. Clone this repo into the workstation from where the rest of this guide will be followed.
+   Move into the directory of this sample.
+```sh
+cd anthos-bm-gcp-terraform/
+```
+
+2. Create a `terraform.tfvars` file from the sample input variables file
+```sh
+cp terraform.tfvars.sample terraform.tfvars
+```
+
+3. Update the `terraform.tfvars` file to include variables specific to your environment
+```sh
+# terraform.tfvars file
+
+project_id       = "<GOOGLE_CLOUD_PROJECT_ID>"
+region           = "<GOOGLE_CLOUD_REGION_TO_USE>"
+zone             = "<GOOGLE_CLOUD_ZONE_TO_USE>"
+credentials_file = "<PATH_TO_GOOGLE_CLOUD_SERVICE_ACCOUNT_FILE>"
+```
+
+4. Add the `mode` variable to the `terraform.tfvars` file
+```sh
+# terraform.tfvars file
+...
+mode             = "install"
+...
+```
+
+5. Navigate to the root directory of this sample and initialize Terraform
+```sh
+terraform init
+```
+
+5. Create a _Terraform_ execution plan
+```sh
+terraform plan
+```
+
+6. Apply the changes described in the _Terraform_ script
+```sh
+terraform apply
+```
+> **Note:** When prompted to confirm the Terraform plan, type 'Yes' and enter
+
+***The `apply` command sets up the Compute Engine VM based bare metal infrastructure. This can take a few minutes (approx. 3-5 mins) for the VMs to get setup.***
+
+---
+### Verify installation
+
+As explained earlier, the Terraform script sets up the GCE VM infrastructure and also
+triggers the installation of Anthos on bare metal on the provisioned VMs. The installation
+is triggered from inside the _admin workstation_ VM.
+
+Upon completion the Terraform script will print the following output to the console.
+```sh
+################################################################################
+#          SSH into the admin host and check the installation progress         #
+################################################################################
+
+> gcloud compute ssh tfadmin@cluster1-abm-ws0-001 --project=<YOUR_PROJECT> --zone=<YOUR_ZONE>
+> tail -f ~/install_abm.log
+
+################################################################################
+```
+
+Use these commands to SSH into the _admin workstation_ and to monitor the installation status.
+Once it is complete the output in the log file will look as follows:
+
+```sh
+...
+[2022-03-22 23:21:12+0000] Moving admin cluster resources to the created admin cluster
+[2022-03-22 23:21:18+0000] Waiting for node update jobs to finish OK
+[2022-03-22 23:22:48+0000] Flushing logs... OK
+[2022-03-22 23:22:48+0000] Deleting bootstrap cluster... OK
+
+Anthos on bare metal installation complete!
+Run [export KUBECONFIG=/home/tfadmin/bmctl-workspace/cluster1/cluster1-kubeconfig] to set the kubeconfig
+Run the [/home/tfadmin/login.sh] script to generate a token that you can use to login to the cluster from the Google Cloud Console
+```
```

Comment:
Suggestion: It took me about 10 minutes to see the following message:
```
Anthos on bare metal installation complete!
```
Perhaps we could add a note about having to wait about 10 minutes.

Reply:
Added a note in 84d54eaaf8bc62376da138f8e7c0b388a02f68af